libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -108,10 +108,14 @@ const AP_Param::GroupInfo AP_ICEngine::var_info[] = {
 
 // constructor
 AP_ICEngine::AP_ICEngine(const AP_RPM &_rpm) :
-    rpm(_rpm),
-    state(ICE_OFF)
+    rpm(_rpm)
 {
     AP_Param::setup_object_defaults(this, var_info);
+
+    if (_singleton != nullptr) {
```

Comment:
This should have been SITL only to be consistent I believe, unless we've changed our minds/existing code on this?

Reply:
on the dev call, Tridge said he wants panics on hardware. It's OK because they only happen on boot

Reply:
@magicrub That's not exactly true. If code is correct then yes, it won't happen after boot. But then, if code is correct that check wouldn't be needed at all...

Reply:
It was discussed on a Dev call last week. Tridge made a comment on the other issue about it. Do we need more discussion?